server/src/main/java/org/elasticsearch/indices/breaker/CircuitBreakerMetrics.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices.breaker;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.telemetry.TelemetryProvider;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A class collecting trip counters for circuit breakers (parent, field data, request, in flight requests and custom child circuit
+ * breakers).
+ *
+ * We decided to have the name as part of the (long) counter metric name instead of being an attribute because aggregating distinct circuit
+ * breakers trip counter values does not make sense, as for instance, summing field_data.trip_count and in_flight_requests.trip_count.
+ * Those counters trip for different reasons even if the underlying reason is "too much memory usage". Aggregating them together results in
+ * losing the ability to understand where the underlying issue is (too much field data?, too many concurrent requests, too large concurrent
+ * requests?). Aggregating each one of them separately to get, for instance, cluster level or cloud region level statistics is perfectly
+ * fine, instead.
+ *
+ * NOTE: here we have the ability to register custom trip counters too. This ability is something a few plugins take advantage of nowadays.
+ * At the time of writing this class it is just "Eql" and "MachineLearning" which track memory used to store "things" that are
+ * application/plugin specific such as eql sequence query objects and inference model objects. As a result, we just have a couple of this
+ * custom counters. This means we have 6 circuit breaker counter metrics per node (parent, field_data, request, in_flight_requests,
+ * eql_sequence and model_inference). We register them a bit differently to keep the ability for plugins to define their own circuit breaker
+ * trip counters.
+ */
+public class CircuitBreakerMetrics {
+    public static final String CIRCUIT_BREAKER_PARENT_TRIP_COUNT_TOTAL = "circuit_breaker.parent.trip_count.total";
```

Comment:
Going to comment on this as it is the first one, but it applies to all of them.
In general, the name looks good! I read the discussion and see you settled for inserting the breaker name in the metric name; it's fair wrt the proposed guidelines, and the comment explain well why.
Suggestions:
- All metrics should start with `es.`
- (optional) `circuit_breaker` is good, maybe just `breaker` could also work - no strong opinion here.
- (optional) `trip_count` is fine, but looks a bit off when paired with `total`. Maybe just `trip.total` would do. Again, I don't feel strongly about it, `_count` just looks redundant to me. If you feel it is clearer this way, don't change.

Reply:
+1 to these suggestions